### PR TITLE
Added extension under existing Kubernetes extension.

### LINKED ISSUE #3: As a user, I would like to be able to use hydrate within the Kubernetes extension
Explore adding this vs code extension under the Kubernetes extension.
Acceptance Criteria: User is able to add extension under existing Kubernetes extension.

### PR DESCRIPTION
**Pull request by madelineliao**
Closes #3 

Edited the extension to work with the existing Kubernetes extension, which displays a user's clusters in the sidebar. Upon right-clicking a cluster, an option to "Hydrate Cluster" is shown in the resulting dropdown, though it currently has no functionality yet.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,36 +1,77 @@
...
-		"vscode": "^1.36.0"
+		"vscode": "^1.35.0"
 	},
 	"categories": [
 		"Other"
 	],
+	"keywords": [
+        "bedrock",
+        "fabrikate",
+		"hydrate",
```
Comment (bnookala) [inline package.json]: indentation

File: package.json [hunk 2]
```diff
@@ -1,36 +1,77 @@
...
+	"extensionDependencies": [
+		"ms-kubernetes-tools.vscode-kubernetes-tools"
 	],
 	"main": "./out/extension.js",
 	"contributes": {
-		"commands": [{
-			"command": "extension.helloWorld",
-			"title": "Hello World"
-		}]
+		"commands": [
```
Comment (bnookala) [inline package.json]: this array is a bit mis-indented

File: package.json [hunk 3]
```diff
@@ -1,36 +1,77 @@
...
+		"typescript": "^3.4.3",
+		"vscode": "^1.1.28"
+	},
+	"dependencies": {
+		"@types/json2md": "^1.5.0",
+		"@types/tmp": "^0.1.0",
+		"deep-object-diff": "^1.1.0",
+		"json2md": "^1.6.3",
+		"shelljs": "^0.8.3",
+		"tmp": "^0.1.0",
```
Comment (bnookala) [inline package.json]: there's no dependency on any of these (yet) except the vscode-kubernetes-tools-api right?

File: src/extension.ts [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
+export async function activate (context: vscode.ExtensionContext) {
+	const clusterExplorerAPI = await k8s.extension.clusterExplorer.v1;
+	const kubectlAPI = await k8s.extension.kubectl.v1;
 
-	// The command has been defined in the package.json file
-	// Now provide the implementation of the command with registerCommand
-	// The commandId parameter must match the command field in package.json
-	let disposable = vscode.commands.registerCommand('extension.helloWorld', () => {
-		// The code you place here will be executed every time your command is executed
+    if (!clusterExplorerAPI.available ||  !kubectlAPI.available) {
```
Comment (bnookala) [inline src/extension.ts]: indentation appears to be off

File: src/extension.ts [hunk 5]
```diff
@@ -1,27 +1,35 @@
...
+        vscode.window.showErrorMessage("Unable to access Kubernetes extension");
+        return;
+    }
 
-		// Display a message box to the user
-		vscode.window.showInformationMessage('Hello World!');
-	});
+    clusterExplorer = clusterExplorerAPI.api;
```
Comment (bnookala) [inline src/extension.ts]: indentation

File: src/extension.ts [hunk 6]
```diff
@@ -1,27 +1,35 @@
...
+	kubectl = kubectlAPI.api;
+	
+	const subscriptions = [
+		vscode.commands.registerCommand('vshydrate.hydrateCluster', hydrateCluster),
+    ];
```
Comment (bnookala) [inline src/extension.ts]: indentation

**Comment by andrebriggs** [comment]
@madelineliao can we say this PR closes the issue #3?  The acceptance criteria is:
> User is able to add extension under existing Kubernetes extension to run hydrate

Since this is currently a no-op maybe we should reword the git issue since we aren't actually running hydrate

**Comment by madelineliao** [comment]
> @madelineliao can we say this PR closes the issue #3? The acceptance criteria is:
> 
> > User is able to add extension under existing Kubernetes extension to run hydrate
> 
> Since this is currently a no-op maybe we should reword the git issue since we aren't actually running hydrate

@andrebriggs  I'll reword the git issue, my understanding is that this issue's goal was to have the extension run under the Kube extension, while the actual Hydrate feature is addressed in #2

**Pull request merged**